### PR TITLE
py-*: add py311

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6.8 21G725 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
Rebased to the last master

**Comment by catap** [comment]
@reneeotten may I ask you also to review this PR?

**Comment by pmetzger** [comment]
FWIW, it looks like this is all pretty innocuous. @reneeotten ?

**Comment by catap** [comment]
@pmetzger it is. And I've build and run `port test ...`for all that ports.

File: python/py-gpsoauth/Portfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ PortGroup           python 1.0
 
 name                py-gpsoauth
 version             1.0.2
-revision            0
+revision            1
```
Comment (reneeotten) [inline python/py-gpsoauth/Portfile:8]: why does this need a `revision` increase? It presumably did build before for the already listed Python versions, but even if  not you are only changing build dependencies here so normally there is no reason to increase the `revision` field.

**Review by reneeotten — COMMENTED** [review]
I've only one question regarding a `revision` increase; other than that it looks good to me

File: python/py-gpsoauth/Portfile (hunk 1 above)
Comment (catap) [reply python/py-gpsoauth/Portfile:8]: I have changed the build backed from `setuptools` to `poetry`, that makes different build dependnecy.

Perhabs it overenginereg and I'll decreased it back.

Comment (reneeotten) [reply python/py-gpsoauth/Portfile]: Exactly, and you don't need a `revision` increase if you only change build dependencies.

**Pull request merged**